07-Extract-NDVI-From-Rasters-in-R.Rmd

```diff
@@ -84,76 +87,78 @@ This lesson is a part of a series of raster data in R lessons:
+lessons. This data is derived from the Landsat satellite and stored in GeoTIFF
+format. Each raster covers the <a href="http://www.neoninc.org/science-design/field-sites/harvard-forest" target="_blank">NEON Harvard Forest field site</a>.  
+If you do not yet have this RasterStack, please create it now. 
 
-To begin, we will use the same raster stack that we used in the previous lesson.
+```{r load-libraries-data }
 
-
-```{r import-ndvi-rasters }
+library(raster)
+library(rgdal)
+library(ggplot2)
 
 # Create list of NDVI file paths
-NDVI_path_HARV <- "Landsat_NDVI/HARV/2011/ndvi"
-all_NDVI_HARV <- list.files(NDVI_path_HARV, full.names = TRUE, pattern = ".tif$")
-
-#view list - note that the full path (relative to our working directory)
-#is included
-all_NDVI_HARV
+NDVI_path <- "Landsat_NDVI/HARV/2011/ndvi"  #assign path to object = cleaner code
+all_NDVI <- list.files(NDVI_path, full.names = TRUE, pattern = ".tif$")
 
 # Create a time series raster stack
-NDVI_stack_HARV <- stack(all_NDVI_HARV)
-
+NDVI_stack <- stack(all_NDVI)
 ```
 
-Once we have created a raster stack, we can efficiently work with our time series.
-
 #Calculate Average NDVI
+Our output goal is a data frame that contains a single, mean NDVI value for each
+raster in our time series. This represents the mean or average NDVI value for
+the area on a given day.  
 
-Our output goal is a dataframe that contains a single, mean NDVI value for each
-raster in our time series. We can calculate the mean for each raster using the
-`cellStats` function. This produces a numeric array of values.
+We can calculate the mean for each raster using the `cellStats` function. This
+produces a numeric array of values. However, we want a data frame so we can 
+convert from the array to a data frame with `as.data.frame()`.
 
 ```{r calculate-avg-NDVI }
 #calculate mean NDVI for each raster
-avg_NDVI_HARV <- cellStats(NDVI_stack_HARV,mean)
+avg_NDVI_HARV <- cellStats(NDVI_stack,mean)
 
-#convert output to data.frame
+#convert output array to data.frame
 avg_NDVI_HARV <- as.data.frame(avg_NDVI_HARV)
 
-#We can do the above two steps with one line of code
-avg_NDVI_HARV <- as.data.frame(cellStats(NDVI_stack_HARV,mean))
+#To be more efficient we could do the above two steps with one line of code
+#avg_NDVI_HARV <- as.data.frame(cellStats(NDVI_stack_HARV,mean))
 
+#view data
 avg_NDVI_HARV
-```
 
-Let's explore our output `data.frame`. Each row has a name corresponding with the
-raster name in our stack
+#view only the value in row 1, column 1 of the data frame
+avg_NDVI_HARV[1,1]
+```
 
-```{r view-dataframe-output }
-#view row names for data frame
-row.names(avg_NDVI_HARV)
+We can see that we now have a data frame with `row.names` based on the original
+file name and a mean NDVI value for each. We can also select values by their
+location in the data frame.
 
-#view the first value in the data.frame
-avg_NDVI_HARV[1,1]
+The column name `avg_NDVI_HARV` for the values is descriptive but doesn't
```

Comment:
Clean this text up please so it's concise and specific. doesn't plot well isn't very specific and the important piece is really the duplication of names which is in parenth.